test/python/quantum_info/operators/symplectic/test_pauli_list.py

```diff
@@ -1999,6 +1999,60 @@ def test_evolve_clifford_qargs(self, phase):
         ]
         self.assertListEqual(value, target)
 
+    def test_group_qubit_wise_commuting(self):
+        """Test grouping qubit-wise commuting operators"""
+
+        with self.subTest(msg="1-qubit PauliList group subops"):
+            # 1-qubit Pauli
+            pauli = PauliList(["I", "X", "Y", "Z"])
+            groups = pauli.group_qubit_wise_commuting()
+            value = [pl.to_labels() for pl in groups]
+            for group in value:
+                group.sort()
+            value.sort()
+            target = [["I", "X"], ["Y"], ["Z"]]
+            self.assertEqual(value, target)
+
+        with self.subTest(msg="2-qubit PauliList group subops"):
+            # 2-qubit Pauli
+            pauli = PauliList(
+                [
+                    "II",
+                    "IX",
+                    "IY",
+                    "IZ",
+                    "XI",
+                    "XX",
+                    "XY",
+                    "XZ",
+                    "YI",
+                    "YX",
+                    "YY",
+                    "YZ",
+                    "ZI",
+                    "ZX",
+                    "ZY",
+                    "iZZ",
+                ]
+            )
+            groups = pauli.group_qubit_wise_commuting()
+            value = [pl.to_labels() for pl in groups]
+            for group in value:
+                group.sort()
+            value.sort()
+            target = [
+                ["II", "IX", "XI", "XX"],
+                ["IY", "XY"],
+                ["IZ", "XZ"],
+                ["YI", "YX"],
+                ["YY"],
+                ["YZ"],
+                ["ZI", "ZX"],
+                ["ZY"],
+                ["iZZ"],
+            ]
```

Comment:
Both these tests assume that the global identity will always get placed in a particular group.  That doesn't appear to be defined behaviour by the function docstring, so is it safe to assume it will always be the case?  By my reading of the docstring, it would still be a valid output if the `"II"` got placed in any of the groups, and the placement in the end is an implementation detail that the graph partition algorithm is greedy (and consequently also dependent on the input order).

Would it be worth testing the behaviour more generally, by something like asserting that every input Pauli appears verbatim somewhere in the output, and for every pair of groups in the output, there is at least one element of one which does not commute with at least one element of the other?  I wouldn't necessarily want the tests to give false negatives if the graph algorithm is changed, but equally, what I'm suggesting is a more complex test, and there's more room for error in the test itself.

Reply:
The way the testing Pauli Lists were written were so that the global identity would end up in a certain group, but it is true that the global identity can be in any group. I see your point with a more complex test, and I can implement that.